src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -207,7 +211,16 @@ private static byte[] BuildRequestHeader(Uri uri, ClientWebSocketOptions options
                 builder.Append("GET ").Append(uri.PathAndQuery).Append(" HTTP/1.1\r\n");
 
                 // Add all of the required headers
-                builder.Append("Host: ").Append(uri.IdnHost).Append(":").Append(uri.Port).Append("\r\n");
+                builder.Append("Host: ");
+                if (string.IsNullOrEmpty(hostHeader))
+                {
+                    builder.Append(uri.IdnHost).Append(":").Append(uri.Port).Append("\r\n");
```

Comment:
Nit: While you're making changes here, consider changing `Append(":")` to `Append(':')`.